activerecord/lib/active_record/relation/calculations.rb

```diff
@@ -389,6 +389,16 @@ def has_include?(column_name)
       def perform_calculation(operation, column_name)
         operation = operation.to_s.downcase
 
+        distinct, column_name = distict_and_column_name(operation, column_name)
+
+        if group_values.any?
+          execute_grouped_calculation(operation, column_name, distinct)
+        else
+          execute_simple_calculation(operation, column_name, distinct)
+        end
+      end
+
+      def distict_and_column_name(operation, column_name)
```

Comment:
Not a fan of this method, but it makes the diff more readable for now.
The logic is quite complex, but I'd like to have separate methods for getting `distinct` and `column_name`.